### PR TITLE
Migrate Windows content from community documentation

### PR DESCRIPTION
**Pull request by GWeck**
See discussion: https://forum.qubes-os.org/t/16193/

### CODE REVIEW

File: user/templates/templates.md
Comment (andrewdavidwong) [inline user/templates/templates.md:93]: This link won't work, because you haven't added this page yet. Did you mean to add more to this PR?

**Comment by andrewdavidwong** [comment]
Would you like me to copy over the existing files from https://github.com/Qubes-Community/Contents/tree/master/docs/os/windows?

**Comment by GWeck** [comment]
I am somewhat at a loss now, probably caused by my low understanding of github. The idea was to get it in three steps:

- PR GWeck:patch-8#209 should add an entry into the table of contents.
- PR GWeck:patch-8#210 should provide a short explanation and a link  in the [Templates ](https://www.qubes-os.org/doc/templates/) introductory text.
- The main files of the community description have been changed, as their internal links were still absolute, pointing to the old position in the community docs. I have modified these links now to be relative so that they should work in the new location. Unfortunately, I created the pull request to add these new files in a different branch (G-Weck:patch-1 instead of GWeck:patch-8).. There I created PR GWeck:patch-1#2 which should provide the corrected files in their new location and thus complete PRs 209 and 210.

I hope I made my intentions clear. Could you have a look into it? Thanks in advance!

**Comment by andrewdavidwong** [comment]
Sorry for the confusion! Let me see if I can help.

**Comment by andrewdavidwong** [comment]
@GWeck, I've pulled your commits from https://github.com/GWeck/qubes-doc/tree/GWeck-patch-1 into this PR. Were these the commits you wanted?

Also, I noticed that you had some image files in this PR. We generally store images in the qubes-attachment repo instead of the qubes-doc repo, so I've taken the liberty of opening a separate PR for adding your images to qubes-attachment (https://github.com/QubesOS/qubes-attachment/pull/61) and merging it. Some of your image files were already present there.

To embed those images in documentation pages, please use this format:

```
[![Image Title](/attachment/doc/image.png)](/attachment/doc/image.png)
```

I didn't change the filenames of the images themselves, just their location.

**Comment by andrewdavidwong** [comment]
@GWeck, I have now also done the following:

- Remove deprecated external redirects
- Add YAML headers
- Fix headings
- Improve naming scheme
- Make permalinks consistent with names
- Fix titles
- Add redirects
- Use preferred heading syntax
- Fix image embeds
- Use relative links for internal links
- Update links

Once the CI log looks okay, we'll try deploying to the testing site and see how it looks.

**Comment by andrewdavidwong** [comment]
@GWeck, would you mind having a look at the "new spellings" that your PR introduced? They're detailed in the log starting at [this line](https://gitlab.com/QubesOS/qubes-doc/-/jobs/3636425872#L453).

**Comment by GWeck** [comment]
Thanks a lot!!!

I had a look at the new files, and they seem to be what I intended. I am just unsure about the permalinks at the beginning which point to the old file names. Is this necessary for the migration, or should they point to the new locations? Currently, the links do not work for me, but maybe this is normal as long as the files are not in the final location - or am I wrong there?

I am closing for now and looking more into it tomorrow - at my location, it's now evening and snowing ...

**Comment by andrewdavidwong** [comment]
> I am just unsure about the permalinks at the beginning which point to the old file names. Is this necessary for the migration, or should they point to the new locations?

You might be looking at the `redirect_from:` entries. Those will redirect from the old URLs and filenames. The `permalink:` values should now be correct.

> Currently, the links do not work for me, but maybe this is normal as long as the files are not in the final location - or am I wrong there?

Currently, this PR has not been merged or deployed to to the staging site yet, so no links are expected to work yet. Let me try to deploy it now.

> I am closing for now and looking more into it tomorrow - at my location, it's now evening and snowing ...

Oh yes, no rush at all! Enjoy your evening, and stay warm! :)

**Comment by andrewdavidwong** [comment]
TestDeploy

**Comment by andrewdavidwong** [comment]
The deployed test site will be available at https://wwwpreview.qubes-os.org/ once it finishes building (shortly after the timestamp of this comment).

File: user/templates/templates.md
Comment (marmarek) [inline user/templates/templates.md:93]: This should point at the page permalink, not md filename. So, `/doc/templates/windows/`

Comment (andrewdavidwong) [reply user/templates/templates.md:93]: Ah, I fixed a ton of these and must have somehow missed this one.

**Comment by andrewdavidwong** [comment]
TestDeploy

**Comment by andrewdavidwong** [comment]
TestDeploy

**Comment by andrewdavidwong** [comment]
Formatting note: Please don't abuse quotation or block quote syntax for "notes." Use `>` only for actual quotations, not to try to get distinctive formatting or styling. (In other words, please respect the separation of content and style.)

If you feel the need to visually distinguish things, consider using alerts instead.

**Comment by andrewdavidwong** [comment]
This link is broken, and I didn't see an obvious replacement for it:

https://github.com/Qubes-Community/Contents/blob/master/docs/customization/windows-gaming-hvm

**Comment by GWeck** [comment]
Not it is looking really good! I fixed the formatting and the broken link.

As far as I can see, PR#210 can now be deployed, too.

**Comment by andrewdavidwong** [comment]
`:warning:` probably won't do what you expect here. I recommend just using standard Markdown instead, like `**Warning:**` or actual HTML alerts.

**Comment by GWeck** [comment]
I changed the formatting so that it now corresponds to that in the community doc.

**Comment by andrewdavidwong** [comment]
Thanks! Could you fix any typos listed [here](https://gitlab.com/QubesOS/qubes-doc/-/jobs/3641298788#L452) when you get a chance? As mentioned above, many items in this list are probably intentional, but at least a couple look like they could be typos, like `Advaced` and `regstry`.

**Comment by GWeck** [comment]
I found some typos -thank you! (Iread it several times, but some still slip through.) Hope I got them all now.

**Comment by andrewdavidwong** [comment]
Thanks! Looks like this is probably ready to undergo review now. Turning it over to @marmarek.

**Comment by andrewdavidwong** [comment]
TestDeploy

**Comment by GWeck** [comment]
Is there anything I could / should do to get this merged into the official documentation?

**Comment by andrewdavidwong** [comment]
> Is there anything I could / should do to get this merged into the official documentation?

No, it is waiting for review.

File: user/templates/windows/windows-qubes-4-1.md
Comment (marmarek) [inline user/templates/windows/windows-qubes-4-1.md:226]: I don't think we should endorse various 3rd-party tools in our official documentation, especially in section "best practices". This _at least_ need some disclaimer that user should evaluate risks themselves, but probably better remove this point.

File: user/templates/windows/windows-qubes-4-1.md
Comment (marmarek) [inline user/templates/windows/windows-qubes-4-1.md:260]: Similar note here - while it might work, there is no way to verify if only benign modification to the image were made. If there is no similar official version, I'd prefer to drop the link and maybe replace with "at your own risk you may use an installation image with bundled updates, but generally we do not recommend this way for security reasons".

File: user/templates/windows/windows-qubes-4-1.md
Comment (marmarek) [inline user/templates/windows/windows-qubes-4-1.md:262]: Mention domains widget instead / in addition to `xentop`?

**Review by marmarek — CHANGES_REQUESTED** [review]

File: user/templates/windows/qubes-windows-tools-4-0.md
Comment (andrewdavidwong) [reply user/templates/windows/qubes-windows-tools-4-0.md:18]: If it's purely about 4.0, then no, IMO.

File: user/templates/windows/windows-qubes-4-0.md
Comment (andrewdavidwong) [reply user/templates/windows/windows-qubes-4-0.md:9]: Since we've removed other (non-Windows) 4.0 documentation, we should be consistent and also do that here, IMO.

File: user/templates/windows/qubes-windows-tools-4-0.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-0.md:18]: Looking at the usage statistics, especially R4.0 going down, I guess the two chapters on R4.0 could be removed. The chapter on migration from R4.0 to R4.1, however, should still be kept for some time, for people mow moving that way.

File: user/templates/windows/migrate-to-4-1.md
Comment (GWeck) [reply user/templates/windows/migrate-to-4-1.md:16]: It's also crazy because of several quirks in Windows. Installing and especially uninstalling software, no matter which one, is normally a nightmare. What is described here is some sort of a way of least resistance but without any guarantee.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:19]: Perhaps just remove the clause `than can be enabled or disabled during installation`.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:20]: Or remove the second sentence in the list title completely?

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:62]: Windows qubes should not be trusted, anyhow. Their installation media is not properly signed, and any patches may be of doubtful quality. So switching off driver signing should not make a big difference.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:115]: O.K. I hope that the hint that no memory balancing is used is clear enough.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:109]: I am somewhat amazed that you had success with just 2 GB of memory. I personally tend to give more if I have it. Windows is slooow otherwise.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:260]: As such a bundle of patches may help considerably to speed up updating, I would prefer to keep it, but with such a cautionary message.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:262]: In addition, just after the last closing parenthesis.

Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:262]: @mamrmarek If it's o.k. with you, I'll change the texts accordingly.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (andrewdavidwong) [reply user/templates/windows/qubes-windows-tools-4-1.md:19]: > Perhaps just remove the clause `than can be enabled or disabled during installation`.

I don't see that clause in the text above.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:20]: See above.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:62]: No change

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:78]: I hope the new text is clearer.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:138]: Some user left this tip in the forum, and I tried it in a standard installation with QWT 4.1-68. It seemed to work and improve the sound quality for Windows 7; for Windows 10 and 11, the sound stayed scratchy.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:253]: This CMD window is necessary for dispVMs, because, without it, the dispVM will close before starting the intended application. The sequence described here must be executed in the AppVM used as a template for the dispVM so that the CMD window is created in each dispVM based on it. I admit that this is ugly, but so far I have not found any way around it.

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:313]: Just added the clause `in dom0`.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:109]: Being cautious, I kept the text as it is. Having enough memory should not be such a problem nowadays, or am I too optimistic there?

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:115]: Added a sentence about disabling memory balancing.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:260]: I added this warning and made it still sharper. But I still think that such help is needed if Windows has to be set up quickly and the security requirements are not too high. Anyhow, I would not trust Windows too much, even as a VM under Qubes.

File: user/templates/windows/windows-qubes-4-1.md
Comment (GWeck) [reply user/templates/windows/windows-qubes-4-1.md:262]: I added the changes now, as commented above, and updated the PR accordingly. I left the two texts on Windows and QWT in Qubes R4.0, but removed the links to them, so the texts themselves are not needed anymore and can be removed if and when appropriate.

**Comment by GWeck** [comment]
With regard to the failed check: These are only words marked in the spelling checker which are regarded as unknown. Unfortunately, I don't know how to instruct this utility to accept them. True misspellings are corrected.

**Comment by andrewdavidwong** [comment]
> With regard to the failed check: These are only words marked in the spelling checker which are regarded as unknown. Unfortunately, I don't know how to instruct this utility to accept them. True misspellings are corrected.

No need to worry. Merging this PR will add your new words to the approved list so that the spell checker won't flag them again in the future. Thank you for fixing the true misspellings!

File: user/templates/windows/qubes-windows-tools-4-1.md
Comment (andrewdavidwong) [inline user/templates/windows/qubes-windows-tools-4-1.md:126]: I just tested these instructions, and I don't see `qrexec_installed` or `guiagent_installed` after installing QWT. The only place where I messed up is that I didn't see this part until it was too late:

> If during installation, the Xen driver requests a reboot, select "No" and let the installation continue - the system will be rebooted later.

Comment (GWeck) [reply user/templates/windows/qubes-windows-tools-4-1.md:126]: I am afraid that if the reboot occurs too early, the installation may be incomplete. The best would probably be to restart it, maybe after first removing what has been installed so far. I never tried what happened there.

**Comment by taradiddles** [comment]
Note-  with the migration of the qubes-community resources to the forum complete, the windows documentation is the only blocker to setting the qubes-community read-only.
@unman - as the current doc maintainer and since you wanted to set the qubes-community repo read-only, any chance you could move this PR forward? It's been 5 months with barely any update.

**Comment by andrewdavidwong** [comment]
It looks like this PR was initially reviewed by @marmarek, then @GWeck made some changes in response to that review, and now it's waiting to be reviewed again. The re-review could be done by @marmarek again or by anyone else who @unman deems qualified (including, potentially, @unman himself). Or @unman could decide no re-review is needed and merge it as is.

**Comment by GWeck** [comment]
O.K. from me - if still anything is needed, just call me.

**Pull request merged**
